integration/integration_test.go

```diff
@@ -1354,6 +1354,45 @@ COPY --from=a /root/date.txt /date.txt`, testImageAlpine, testImageAlpine),
 	})
 }
 
+func TestEmbedBinaryImage(t *testing.T) {
+	t.Parallel()
+
+	srv := createGitServer(t, gitServerOptions{
+		files: map[string]string{
+			".devcontainer/Dockerfile": fmt.Sprintf("FROM %s\nRUN date --utc > /root/date.txt", testImageAlpine),
+			".devcontainer/devcontainer.json": `{
+			"name": "Test",
+			"build": {
+				"dockerfile": "Dockerfile"
+			},
+		}`,
+		},
+	})
+
+	testReg := setupInMemoryRegistry(t, setupInMemoryRegistryOpts{})
+	testRepo := testReg + "/test"
+	ref, err := name.ParseReference(testRepo + ":latest")
+	require.NoError(t, err)
+
+	_, err = runEnvbuilder(t, options{env: []string{
+		envbuilderEnv("GIT_URL", srv.URL),
+		envbuilderEnv("CACHE_REPO", testRepo),
+		envbuilderEnv("PUSH_IMAGE", "1"),
+	}})
+	require.NoError(t, err)
+
+	_, err = remote.Image(ref)
+	require.NoError(t, err, "expected image to be present after build + push")
+
+	ctr, err := runEnvbuilder(t, options{env: []string{
+		envbuilderEnv("FALLBACK_IMAGE", ref.String()),
+	}})
+	require.NoError(t, err)
```

Comment:
What we really want to do is the equivalent of 

```
docker run -it --rm localhost:5000/test-embed-binary-image:latest /bin/sh -c '[[ -f /.envbuilder/bin/envbuilder ]] && echo exists'
```

What you have here actually is more similar to:

```
docker run -it --rm -e FALLBACK_IMAGE=localhost:5000/test-embed-binary-image:latest /bin/sh -c '[[ -f /.envbuilder/bin/envbuilder ]] && echo exists'
```

The `envbuilder` binary that this is testing is from having run envbuilder itself, and not from the image that was pushed to the local registry.